web/static/js/reducers/guest.js

```diff
@@ -18,18 +21,31 @@ export default (state = initialState, action) => {
   }
 }
 
+const validEmail = (email) => {
+  return /^[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*@(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?$/.test(email)
+}
+
 const changeEmail = (state, action) => {
   return {
     ...state,
-    email: action.email
+    data: {
+      ...state.data,
+      email: action.email
+    },
+    errors: {
+      ...state.errors,
+      email: !validEmail(action.email)
+    }
   }
 }
 
 const changeLevel = (state, action) => {
   if (['reader', 'editor'].includes(action.level)) {
+    const newData = {...state.data}
```

Comment:
Here you should avoid defining a `const` and use a spread to define the new data